### PR TITLE
#214 :wrench: Improve ComponentType and OrchestrationType enums

### PR DESCRIPTION
**Pull request by dlicheva**
[Ticket](https://github.com/ocurity/dracon/issues/214)
Used [go-enum](https://github.com/abice/go-enum/) to generate types for:
- orchestrationType - converted to a string enum, with prefix, because "Unknown" is a duplicate with the componentType
- componentType - converted to a string enum, without prefix because the types are more widely used outside the components package

To generate the enums I used:
- orchestrationType
```
docker run -w /app -v $(pwd):/app abice/go-enum --file /app/pkg/components/types.go --marshal --lower --ptr --mustparse --sqlnullstr --sql --names -b components
```
- componentType
```
docker run -w /app -v $(pwd):/app abice/go-enum --file /app/pkg/components/metadata.go --marshal --mustparse --sqlnullstr --sql --names --values --noprefix -b components
```

### CODE REVIEW

File: pkg/components/metadata.go [hunk 1]
```diff
@@ -116,51 +34,12 @@ func LabelValueOneOf(labels map[string]string, componentTypes ...ComponentType)
...
-
-// UnmarshalText unmarshalls bytes into a `ComponentType`
-func (ct *ComponentType) UnmarshalText(text []byte) error {
-	text = bytes.Trim(bytes.Trim(text, `"`), ` `)
-	parsedComponent, err := ToComponentType(string(text))
-	if err == nil {
-		*ct = parsedComponent
+// MustGetComponentType converts a string into a ComponentType enum value or
+// panics
+func MustGetComponentType(cts string) ComponentType {
```
Comment (ptzianos) [inline pkg/components/metadata.go:39]: I don't think we need this any more, the tool has produced a `MustParseComponentType` that does the same thing

**Comment by ptzianos** [comment]
@dlicheva re componentType being an int enum, I see two solutions, either we leave it as is, or we convert it into a string enum and add a `LessThan` and a `Previous` function in the metadata package that responds to external user any queries about ordering

**Comment by dlicheva** [comment]
Thanks @ptzianos ! I've updated these, so they both have string enums. I've also added some quick documentation.

**Comment by ptzianos** [comment]
@dlicheva a couple of nits and it's ready to be merged

File: pkg/components/metadata.go [hunk 2]
```diff
@@ -1,98 +1,17 @@
 package components
 
 import (
-	"bytes"
-	"fmt"
-
 	"github.com/go-errors/errors"
+	"slices"
```
[the remark this answers is not in the document]
Comment (dlicheva) [reply pkg/components/metadata.go:5]: Weird... my IDE automatically reorganizes them so slices is in the end

**Pull request merged**